gms/gossiper.cc

```diff
@@ -1064,45 +1064,49 @@ future<> gossiper::replicate(inet_address ep, application_state key, const versi
     });
 }
 
-future<> gossiper::advertise_removing(inet_address endpoint, utils::UUID host_id, utils::UUID local_host_id) {
-    return seastar::async([this, g = this->shared_from_this(), endpoint, host_id, local_host_id] {
-        auto& state = get_endpoint_state(endpoint);
-        // remember this node's generation
-        int generation = state.get_heart_beat_state().get_generation();
-        logger.info("Removing host: {}", host_id);
-        auto ring_delay = std::chrono::milliseconds(_cfg.ring_delay_ms());
-        logger.info("Sleeping for {}ms to ensure {} does not change", ring_delay.count(), endpoint);
-        sleep(ring_delay).get();
-        // make sure it did not change
-        auto& eps = get_endpoint_state(endpoint);
-        if (eps.get_heart_beat_state().get_generation() != generation) {
-            throw std::runtime_error(format("Endpoint {} generation changed while trying to remove it", endpoint));
-        }
+future<> gossiper::advertise_removing(inet_address endpoint, utils::UUID host_id, utils::UUID local_host_id, abort_source& as) {
+    return seastar::with_gate(_gate, [this, g = this->shared_from_this(), endpoint, host_id, local_host_id, &as] {
```

Comment:
You can use a half indentation to reduce the changes and restore the indentation later.